### PR TITLE
Reorganize compute docs and add sample code

### PR DESCRIPTION
**Pull request by goderbauer**
Adds an example to `compute` and puts the main docs on `compute` since that's where people end up when they click-through on `compute` in their IDE. Previously, the main docs where on `ComuteImpl`, which is just an implementation detail and the docs there weren't super-discoverable.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/isolates.dart [hunk 1]
```diff
@@ -21,38 +21,78 @@ typedef ComputeCallback<Q, R> = FutureOr<R> Function(Q message);
...
-/// A function that spawns an isolate and runs a callback on that isolate. This
-/// method should be used to execute parallel tasks that reduce the risk of
-/// skipping frames.
-///
-/// {@youtube 560 315 https://www.youtube.com/watch?v=5AxWC49ZMzs}
-///
 /// See also:
 ///
-///   * [ComputeImpl], for function parameters and usage details.
+///   * [ComputeImpl], for the [compute] function's signature.
```
Comment (mit-mit) [inline packages/flutter/lib/src/foundation/isolates.dart:97]: Isn't this rather `the [compute] function's implementation.` ?

Comment (goderbauer) [reply packages/flutter/lib/src/foundation/isolates.dart:97]: `ComputeImpl` is just a typedef specifying the signature of the compute function. The name is maybe a little unfortunate...

File: packages/flutter/lib/src/foundation/isolates.dart [hunk 2]
```diff
@@ -11,48 +11,89 @@ import '_isolates_io.dart'
...
+/// Qualifying functions include:
+///
+///   * top-level functions
+///   * static methods
+///   * closures that only capture objects that can be sent to an isolate
+///
+/// Using closures must be done with care. Due to
+/// [dart-lang/sdk#36983](https://github.com/dart-lang/sdk/issues/36983) a
+/// closure may captures objects that - while not directly used in the closure
+/// itself - may prevent it from being sent to an isolate.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/foundation/isolates.dart:82]:
```suggestion
/// closure may captures objects that, while not directly used in the closure
/// itself, may prevent it from being sent to an isolate.
```

The dashes work, but commas are more grammatically correct, IMO.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/foundation/isolates.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/src/foundation/isolates.dart:82]: Sounds good. Commas it is!

**Pull request merged**